tensorflow/python/keras/benchmarks/benchmark_util.py

```diff
@@ -108,19 +110,26 @@ def measure_performance(model_fn,
   avg_epoch_time_list, wall_time_list, exp_per_sec_list = [], [], []
   total_num_examples = epochs * num_examples
 
+  strategy = distribution_util.get_distribution_strategy(
+      distribution_strategy=distribution_strategy,
+      num_gpus=num_gpus)
+
   for _ in range(run_iters):
     timer = timeit.default_timer
     t0 = timer()
-    model = model_fn()
-    build_time = timer() - t0
-
-    t1 = timer()
-    model.compile(
-        optimizer=optimizer,
-        loss=loss,
-        metrics=metrics,
-    )
-    compile_time = timer() - t1
+    # Init the distribution strategy scope for each iteration.
+    strategy_scope = distribution_util.get_strategy_scope(strategy)
+    with strategy_scope:
+      model = model_fn()
+      build_time = timer() - t0
+
+      t1 = timer()
+      model.compile(
```

Comment:
Does compile need to be within strategy scope? IIUC, putting model_fn into it should work.

Reply:
We should place `complie()` into the scope. I checked with [this post](https://keras.io/guides/distributed_training/#singlehost-multidevice-synchronous-training)  
```
with strategy.scope():
  # Everything that creates variables should be under the strategy scope.
  # In general this is only model construction & `compile()`.
  model = Model(...)
  model.compile(...)
```

Reply:
Yeah, that's right! Both might have variable creations... I got the incorrect expression.